sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -131,8 +136,41 @@ internal class ViewAnnotationManagerImpl(
     viewUpdatedListenerSet.remove(listener)
   }
 
+  @AnyThread
+  override fun setViewAnnotationUpdateMode(mode: ViewAnnotationUpdateMode) {
+    renderThread.viewAnnotationMode = mode
+  }
+
+  @AnyThread
+  override fun getViewAnnotationUpdateMode(): ViewAnnotationUpdateMode {
+    return renderThread.viewAnnotationMode
+  }
+
+  /**
+   * First call will be always from Mapbox render thread that will be ALWAYS followed up
+   * another call from Android main thread.
+   */
+  @AnyThread
   override fun onViewAnnotationPositionsUpdate(positions: MutableList<ViewAnnotationPositionDescriptor>) {
-    drawAnnotationViews(positions)
+    // When called from render thread it means we're in the end of [MapInterface.render] call if positions did change.
```

Comment:
Here I'd just left indication which if branch happens first, and which goes second